L1Trigger/Phase2L1ParticleFlow/plugins/L1MetPfProducer.cc

```diff
@@ -48,34 +52,80 @@ class L1MetPfProducer : public edm::global::EDProducer<> {
   const int invTableBits_ = 10;
   const int invTableSize_ = (1 << invTableBits_);
 
+  // hls4ml emulator objects
+  bool useMlModel_;
+  std::shared_ptr<hls4mlEmulator::Model> model;
+  std::string modelVersion_;
+  typedef ap_fixed<32, 16> input_t;
+  typedef ap_fixed<32, 16> result_t;
+  int numContInputs_ = 4;
+  int numPxPyInputs_ = 2;
+  int numCatInputs_ = 2;
+  int numInputs_ = numContInputs_ + numPxPyInputs_ + numCatInputs_;
+
   void Project(pt_t pt, phi_t phi, pxy_t& pxy, bool isX, bool debug = false) const;
   void PhiFromXY(pxy_t px, pxy_t py, phi_t& phi, bool debug = false) const;
 
+  int EncodePdgId(int pdgId) const;
+
   void CalcMetHLS(std::vector<float> pt, std::vector<float> phi, reco::Candidate::PolarLorentzVector& metVector) const;
+  void CalcMlMet(std::vector<float> pt,
+                 std::vector<float> eta,
+                 std::vector<float> phi,
+                 std::vector<float> puppiWeight,
+                 std::vector<int> pdgId,
+                 std::vector<int> charge,
+                 reco::Candidate::PolarLorentzVector& metVector) const;
```

Comment:
Is it really necessary to copy the vectors and pass to CalcMlMet (or CalcMetHLS). Can one not pass the const reference to the vectors?